sources/packages/api/src/services/application/application.service.ts

```diff
@@ -950,4 +955,52 @@ export class ApplicationService extends RecordDataModelService<Application> {
       ) <= COE_WINDOW
     );
   }
+  /**
+   * Deny the Confirmation Of Enrollment(COE) for an Application.
+   * Updates only applications that have the COE status as required.
+   * @param applicationId application id to be updated.
+   * @param locationId location that is setting the offering.
+   * @param coeDeniedReasonId COE Denied reason id for a student application.
+   * @param otherReasonDesc when other is selected as a COE denied reason, text for the reason
+   * is populated.
+   * @returns updated application.
+   */
+  async setDeniedReasonForConfirmationOfEnrollment(
+    applicationId: number,
+    locationId: number,
+    coeDeniedReasonId: number,
+    otherReasonDesc?: string,
+  ): Promise<Application> {
+    const application = await this.repo.findOne({
+      id: applicationId,
+      location: { id: locationId },
+      coeStatus: COEStatus.required,
+      applicationStatus: Not(
+        In([
+          ApplicationStatus.completed,
+          ApplicationStatus.overwritten,
+          ApplicationStatus.cancelled,
+        ]),
+      ),
+    });
+    if (!application) {
+      throw new CustomNamedError(
+        "Not able to find an application that requires a COE to be completed.",
+        COE_REQUEST_NOT_FOUND_ERROR,
+      );
+    }
+
+    application.coeDeniedReason = {
+      id: coeDeniedReasonId,
+    } as COEDeniedReason;
+    if (COE_DENIED_REASON_OTHER_ID === coeDeniedReasonId && !otherReasonDesc) {
+      throw new CustomNamedError(
+        "Other is selected as COE reason, specify the reason for the COE denial.",
+        COE_DENIED_REASON_NOT_FOUND_ERROR,
```

Comment:
I noticed that this constant is created in a common/generic constant file and I am not sure what we are following here. The idea is to have all the constants under the same file because they are constants?
I would recommend creating the constants in a context where they are more meaningful.
For instance, if the idea is not to have the constants in the same service class where it will be mainly used, we can create a const file for that service.

Reply:
yes @andrewsignori-aot , the idea was to add all constants in the `utils/constants` file. right @guru-aot ?

Reply:
It does not seem right to me. If there is a constant that is globally used and it is generic, maybe, if there is not a specific file to add, we can add to a common/generic/altogether constants file, otherwise, why not create the constants near the folder/context that they belong to?

Reply:
> yes @andrewsignori-aot , the idea was to add all constants in the `utils/constants` file. right @guru-aot ?

As we talked we are not going to add a constant to a generic constants file just because it is a constant.
We already have some other constant files like database/constants.ts.
@ann-aot and I were talking and we can have a constant file to keep values that represent configurations that are not going to be exposed as environment variables but could be changed in the future, like the ones already in the file, as shown below. In this way, if the Ministry is looking to do a change in some hardcoded configurations they will be in the same place. We can rename the constant file to something like system-configuration-constants.ts
@guru-aot, @dheepak-aot, @ann-aot, does it make sense?

![image](https://user-images.githubusercontent.com/61259237/132595146-a58c3ae3-c198-4f0d-922d-3501c9b7e015.png)

Reply:
`system-configuration-constants.ts`, `configurable-constants.ts` or `application-configuration-constant.ts` make sense to me

Reply:
I agree with even for my existing story, I am creating a utilities constant where I am creating a Intitution_type for BC_Private in sources/packages/web/src/constants/utilities.ts. This is a separate file and is created for web package level.

Reply:
@andrewsignori-aot, I expressed same(similar) opinion in other comment. We can functionally/technically separate the constants file to make it more modular instead of having in same file. 
 
e.g
http-status-code.ts (technical)
application-constants.ts(functional)